tests/rptest/tests/acls_test.py

```diff
@@ -282,24 +293,40 @@ def test_mtls_principal(self, rules=None, fail=False):
             except ClusterAuthorizationError:
                 assert fail, "list acls should have succeeded"
 
+    @cluster(num_nodes=3, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    def test_migrate_cluster_to_mtls(self):
+
+        # Start cluster with tls certs and sasl and everything else default.
+        self.prepare_cluster(use_tls=True, use_sasl=True)
+
+        # Validate perms
+        self.check_permissions()
+
+        # Must restart all nodes in order to change broker level configs
+        for node in self.redpanda.nodes:
+            self.redpanda.stop_node(node)
+
+        # Adding authz, authn method, and PMRs
+        self.security.kafka_enable_authorization = True
+        self.security.endpoint_authn_method = 'mtls_identity'
```

Comment:
Nit: Should we add a comment that says these two settings override `enable_sasl`?

Reply:
Done.